modules/fundamental/src/purl/service/mod.rs

```diff
@@ -249,6 +273,25 @@ impl PurlService {
         }
     }
 
+    pub async fn purls_by_uuid<C: ConnectionTrait>(
+        &self,
+        uuids: Vec<Uuid>,
+        deprecation: Deprecation,
+        connection: &C,
+    ) -> Result<Vec<PurlDetails>, Error> {
+        let items = qualified_purl::Entity::find()
+            .filter(qualified_purl::Column::Id.is_in(uuids))
+            .all(connection)
+            .await?;
+
+        let mut details = Vec::with_capacity(items.len());
+        for purl in items {
+            details
+                .push(PurlDetails::from_entity(None, None, &purl, deprecation, connection).await?);
+        }
+        Ok(details)
+    }
```

Comment:
Same applies to this one as my comment above.

Reply:
done